### PR TITLE
MyPy Setup + CI Rework

### PR DESCRIPTION
**Pull request by hatchet-temporary**
The vast majority of the line diff here is formatting changes from `black`, which are a one-off for this PR. To review: mostly need to look at the changes in `hatchet.py` and the config (yaml/toml) files, everything else is formatting

* Setting up a pre-commit hook for MyPy (mostly for backwards compat)
* Updating CI to run the linters (black, isort, mypy, soon to be ruff) on all files every time - takes <30sec in total
* Fixing type issues in `hatchet.py` 

The plan here will be to incrementally add more files to MyPy as we work on them, so we don't need to try to fix all of the errors in one shot, and then hopefully get to a state in a few weeks where we can check the whole codebase successfully 🥳 

Tested by running a couple examples on local

### CODE REVIEW

File: hatchet_sdk/contracts/events_pb2.pyi [hunk 1]
```diff
@@ -1,13 +1,27 @@
-from google.protobuf import timestamp_pb2 as _timestamp_pb2
```
Comment (hatchet-temporary) [inline hatchet_sdk/contracts/events_pb2.pyi:1]: Are these stubs autogenerated? I was assuming running the formatters over them is fine, but just wanted to confirm

File: hatchet_sdk/contracts/dispatcher_pb2.pyi [hunk 2]
```diff
@@ -1,9 +1,16 @@
+# type: ignore
+
+from typing import ClassVar as _ClassVar
```
Comment (grutt) [inline hatchet_sdk/contracts/dispatcher_pb2.pyi:3]: did you add these yourself, these are autogenerated so it'll be overwritten on next gen

File: hatchet_sdk/contracts/events_pb2.pyi (hunk 1 above)
Comment (grutt) [reply hatchet_sdk/contracts/events_pb2.pyi:1]: they are, probably answers my q above.

File: hatchet_sdk/hatchet.py [hunk 3]
```diff
@@ -47,40 +58,43 @@ def inner(cls) -> WorkflowMeta:
...
     name: str = "",
     timeout: str = "",
-    parents: List[str] | None = None,
+    parents: list[str] | None = None,
     retries: int = 0,
-    rate_limits: List[RateLimit] | None = None,
-    desired_worker_labels: dict[str:DesiredWorkerLabel] = {},
-):
+    rate_limits: list[RateLimit] | None = None,
+    desired_worker_labels: dict[str, DesiredWorkerLabel] = {},
```
Comment (grutt) [inline hatchet_sdk/hatchet.py:74]: 😭 python typing is so hard ... why didn't this complain!

**Review by grutt — APPROVED** [review]

File: hatchet_sdk/hatchet.py (hunk 3 above)
Comment (hatchet-temporary) [reply hatchet_sdk/hatchet.py:74]: 😂 yeaahh....

File: hatchet_sdk/contracts/dispatcher_pb2.pyi (hunk 2 above)
Comment (hatchet-temporary) [reply hatchet_sdk/contracts/dispatcher_pb2.pyi:3]: cool, sounds good - these should be safe then 👍

Comment (hatchet-temporary) [reply hatchet_sdk/contracts/dispatcher_pb2.pyi]: alright just excluded those files so the diff is a lot smaller!

**Pull request merged**